target/getmail/getmail-service.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# shellcheck source=../scripts/helpers/log.sh
+source /usr/local/bin/helpers/log.sh
+
+# Directory, where "oldmail" files are stored.
+# getmail stores its state - its "memory" of what it has seen in your POP/IMAP account - in the oldmail files.
+GETMAIL_DIR=/var/lib/getmail
+
+# Kill all child processes on EXIT.
+# Otherwise 'supervisorctl restart getmail' leads to zombie 'sleep' processes.
+trap 'pkill --parent $$' EXIT
+
+function _syslog_error() {
+  logger --id --priority mail.err -t getmail6 "$*"
+}
```

Comment:
Do we need any context about `logger`? I guess the function name is sufficient, and both you and I know about what the command itself is doing.

- I think the `-t` can benefit from the long option name.
- `--id` will append a number which I assume is the PID of the `logger` if an explicit one is not provided, so within the same script two consecutive `logger` commands may just increment the PID if there's nothing else going on concurrently. We can omit that since that's not providing any value as-is 🤔 
- The `$*` if I understand correctly is to treat all args as a single value, as opposed to `$@`? Would that be relevant vs `$1`? (_clearer expectation of input_)

```suggestion
function _syslog_error() {
  logger --priority mail.err --tag getmail6 "${1}"
}
```

Related review comment: https://github.com/docker-mailserver/docker-mailserver/pull/4156#discussion_r1717825957

Reply:
> Would it be worth adding any additional context about logger for maintenance?

If you have ideas regarding comments, I'll add them. I don't see a need for it. Everyone can look up commands unknown to them :)

> I think the -t can benefit from the long option name.

I wanted to go with the long options, I just forgot here. I'll fix that.

> --id will append a number which I assume is the PID ....

I've added the `--id` here, because all other services log also with their pids. I will however change that to `--id $$` , so it contains the PID of `getmail-service.sh`.

> The $* if I understand correctly is to treat all args as a single value, as opposed to $@? Would that be relevant vs $1? (clearer expectation of input)

Yeah. It makes no difference in that wether we use `$1` or `$*`. `$@` is `$*` but with the arguments quoted (not suitable here). I'll change it to `$1` to make it more clear.

Reply:
I've edited my answer above.